erpnext/controllers/subcontracting_controller.py

```diff
@@ -89,6 +90,13 @@ def set_items_conversion_factor(self):
 				item.conversion_factor = 1
 
 	def validate_items(self):
+		table = DocType("Purchase Order Item")
+		query = (
+			frappe.qb.from_(table)
+					 .select(table.name, table.qty, table.sco_qty, table.fg_item_qty)
+					 .where(table.parent == self.purchase_order)
+		)
+		po_items = query.run(as_dict=True)
```

Comment:
- fg_item_qty is not using so no need to select

- Make a dict po_wise_qty = {"purchase_order_item": "qty - sco_qty"} and use it instead of the below code

```
po_item = next(po_item for po_item in po_items if po_item.name == item.purchase_order_item)
if self.doctype not in "Subcontracting Receipt" and item.qty > ((po_item.qty - po_item.sco_qty) / item.sc_conversion_factor):
```

use

```
sco_qty = po_wise_qty.get(item.purchase_order_item)
if self.doctype not in "Subcontracting Receipt" and item.qty > (flt(sco_qty) / item.sc_conversion_factor)
```